### PR TITLE
Capi Multiple

### PR DESCRIPTION
**Pull request by Ap0c**
# In This PR:

- cAPI Multiple Paidfor
- cAPI Multiple Supported
- Some supporting SASS files migrated from frontend.
- Some supporting SVGs migrated from frontend.
- A shared `js` file containing most of the code for the two templates.

# Pending:

- Tested in PROD behind ad cookies, but will wait for @regiskuckaertz 's resize listener PR before merging, as there's still minor cropping on window resize.

@annebyrne @JonNorman @Calum-Campbell

### CODE REVIEW

File: src/_shared/scss/_brandbadge.scss [hunk 1]
```diff
@@ -0,0 +1,145 @@
+.badge {
```
Comment (annebyrne) [inline src/_shared/scss/_brandbadge.scss]: Is this necessary? In the CAPI single the badge style works without this file.

File: src/_shared/scss/_core.scss [hunk 2]
```diff
@@ -182,8 +190,17 @@ $mq-breakpoints: (
 
 .justify--start { justify-content: flex-start; }
 
-// ----- Gimbap ----- //
+// Gimbap fonts
 
+@mixin f-headline {
```
Comment (annebyrne) [inline src/_shared/scss/_core.scss]: I believe Regis has removed all the font mixins here: https://github.com/guardian/commercial-templates/commit/88c509e9d4ffcd1ec81713f30a2591f1c801709c

File: src/_shared/svgs/camera.svg [hunk 3]
```diff
@@ -0,0 +1 @@
+<svg width="18" height="13" viewBox="0 0 18 13"><path d="M18 3.5v8l-1.5 1.5h-15l-1.5-1.5v-8l1.5-1.5h3.5l2-2h4l2 2h3.5l1.5 1.5zm-9 7.5c1.9 0 3.5-1.6 3.5-3.5s-1.6-3.5-3.5-3.5-3.5 1.6-3.5 3.5 1.6 3.5 3.5 3.5z"/></svg>
```
Comment (annebyrne) [inline src/_shared/svgs/camera.svg]: The SVG files should be present on master following this merge: https://github.com/guardian/commercial-templates/commit/ff0b235fc48c6d8aee025331dab74c843ed0ad0c

File: src/_shared/scss/_popup.scss [hunk 4]
```diff
@@ -0,0 +1,297 @@
+/* ==========================================================================
```
Comment (annebyrne) [inline src/_shared/scss/_popup.scss]: Is this file needed? See Regis' comment: https://github.com/guardian/commercial-templates/pull/43

File: src/_shared/js/capi-multiple.js [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+		let advertRow = document.querySelector('.adverts__row');
+		advertRow.appendChild(cardList);
+		editionLink(cardsInfo.articles[0].edition, isPaid);
+
+	};
+
+}
+
+// Displays or hides a popup.
+function togglePopup (button, popup) {
```
Comment (annebyrne) [inline src/_shared/js/capi-multiple.js]: Could this be replaced with EnableToggle() from the /_shared/js/ui.js file?

File: src/_shared/scss/_core.scss [hunk 6]
```diff
@@ -10,9 +10,17 @@ $gs-column-width: 60px;
 $gs-row-height: 36px;
 $gs-max-columns: 16;
 
+// For popups.
```
Comment (annebyrne) [inline src/_shared/scss/_core.scss]: Following earlier comments about popup and branding styling, are these additional styles needed?

File: src/_shared/scss/_brandbadge.scss (hunk 1 above)
Comment (Ap0c) [inline src/_shared/scss/_brandbadge.scss]: Yeah it's used for the 'Supported by' tagline in the supported ad.

File: src/_shared/scss/_core.scss (hunk 2 above)
Comment (Ap0c) [inline src/_shared/scss/_core.scss]: Ah good catch. Some of the mixins he removed were being used for the multiple, so I merged those changes and went through replacing all the instances. Looks like I missed that `f-headline`!

File: src/_shared/scss/_popup.scss (hunk 4 above)
Comment (Ap0c) [inline src/_shared/scss/_popup.scss]: Looks like there was a single useful property being taken from the entire file, so I copied it to index and dropped the file.

File: src/_shared/scss/_core.scss (hunk 6 above)
Comment (Ap0c) [inline src/_shared/scss/_core.scss]: Nope! 😄

File: src/_shared/js/capi-multiple.js (hunk 5 above)
Comment (Ap0c) [inline src/_shared/js/capi-multiple.js]: 507dbbe

File: src/_shared/js/capi-multiple.js [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+}
+
+export default function capiMultiple (adType) {
+
+	const isPaid = (adType === 'paidfor');
+
+	getIframeId()
+	.then(retrieveCapiData)
+	.then(capiData => buildCards(capiData, isPaid))
+	.then(write)
```
Comment (regiskuckaertz) [inline src/_shared/js/capi-multiple.js]: I feel this style is a bit dangerous in this case because there are other things happening in `buildCards` which are not explicit here

File: src/_shared/scss/_brandbadge.scss (hunk 1 above)
Comment (regiskuckaertz) [inline src/_shared/scss/_brandbadge.scss]: Actually the GLABS team changed `advert__badge` to `badge` last week. We can probably get rid of `advert__badge` ourselves and use this one instead.

File: src/_shared/scss/_brandbadge.scss [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+    .badge--alt {
+        @include mq(tablet) {
+            float: left;
+            clear: left;
+            border-top: 1px dotted $neutral-5;
+        }
+    }
+}
+
+.content__main-column--article,
```
Comment (regiskuckaertz) [inline src/_shared/scss/_brandbadge.scss]: Of course you  can get rid of a 99% of this file.

File: src/_shared/svgs/video-icon.svg [hunk 9]
```diff
@@ -0,0 +1 @@
+<svg width="36" height="23" viewBox="0 0 36 23"><path d="M3.2 0l-3.2 3.3v16.4l3.3 3.3h18.7v-23h-18.8m30.4 1l-8.6 8v5l8.6 8h2.4v-21h-2.4"/></svg>
```
Comment (regiskuckaertz) [inline src/_shared/svgs/video-icon.svg]: Duplicates?

File: src/_shared/svgs/volume-high.svg [hunk 10]
```diff
@@ -0,0 +1 @@
+<svg width="16" height="13" viewBox="0 0 16 13"><path d="M3 4h-2l-1 1v3l1 1h2l4 4h1v-13h-1l-4 4zm11.7 2.5c0 2-.7 3.8-1.8 5.2l.4.4c1.6-1.3 2.6-3.3 2.6-5.6s-1-4.3-2.6-5.6l-.4.4c1.2 1.5 1.8 3.3 1.8 5.2m-3.7 0c0 1.1-.3 2.2-.9 3.1l.5.5c.8-1 1.4-2.2 1.4-3.6s-.6-2.6-1.5-3.5l-.5.5c.6.8 1 1.9 1 3"/></svg>
```
Comment (regiskuckaertz) [inline src/_shared/svgs/volume-high.svg]: Duplicates?

File: src/capi-multiple-supported/index.scss [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+@import '_brandbadge';
+
+// For cards 3 and 4.
+.hide-until-mobile-landscape {
+    @include mq($until: mobileLandscape) {
+        display: none !important;
+    }
+}
+
+a {
```
Comment (regiskuckaertz) [inline src/capi-multiple-supported/index.scss]: You probably want to use `blink` and `blink__anchor` here

File: src/_shared/js/capi-multiple.js [hunk 12]
```diff
@@ -0,0 +1,262 @@
...
+	title.appendChild(headline);
+
+}
+
+// Constructs the title part of the card: headline and media icon.
+function buildTitle (card, cardInfo, cardNumber) {
+
+	let title = card.querySelector('.advert__title');
+
+	if (cardInfo.videoTag) {
```
Comment (Ap0c) [inline src/_shared/js/capi-multiple.js]: We discussed whether to do that when working on the endpoint. The previous endpoint used booleans and I think @annebyrne  discovered that it was simpler to stick with that in the controller code.

Comment (Ap0c) [inline src/_shared/js/capi-multiple.js]: Relevant commit in frontend [here](https://github.com/guardian/frontend/pull/14473/commits/3f6e7fcf9a118adda6fa99496af9b5c54ea93ca8).

File: src/_shared/js/capi-multiple.js [hunk 13]
```diff
@@ -0,0 +1,262 @@
...
+	brandLogo: '[%BrandLogo%]'
+};
+
+// Loads the card data from CAPI in JSON format.
+function retrieveCapiData () {
+
+	let params = new URLSearchParams();
+	params.append('k', '[%SeriesURL%]');
+
+	OVERRIDES.urls.map(url => {
```
Comment (Ap0c) [inline src/_shared/js/capi-multiple.js]: Yup, you're right: 2dada9a.

File: src/_shared/js/capi-multiple.js [hunk 14]
```diff
@@ -0,0 +1,262 @@
...
+		return sources;
+
+	}, sourcesFragment);
+
+}
+
+// Insert between IE comments, a pain.
+function insertBetweenComments (card, sources) {
+
+	let picture = card.querySelector('picture');
```
Comment (Ap0c) [inline src/_shared/js/capi-multiple.js]: 667b979

File: src/_shared/js/capi-multiple.js (hunk 7 above)
Comment (Ap0c) [inline src/_shared/js/capi-multiple.js]: I've just remembered why I included `linkEdition` here, it's because it needs access to the `edition` value from cAPI; the same is true for `addBranding`. I've renamed the function to make it clearer what it's doing.

File: src/_shared/scss/_brandbadge.scss (hunk 8 above)
Comment (Ap0c) [inline src/_shared/scss/_brandbadge.scss]: 9343715

File: src/_shared/svgs/video-icon.svg (hunk 9 above)
Comment (Ap0c) [inline src/_shared/svgs/video-icon.svg]: 23ff6ca

File: src/_shared/svgs/camera.svg (hunk 3 above)
Comment (Ap0c) [inline src/_shared/svgs/camera.svg]: 23ff6ca

File: src/capi-multiple-supported/index.scss (hunk 11 above)
Comment (Ap0c) [inline src/capi-multiple-supported/index.scss]: 3b541d9

File: src/_shared/scss/_brandbadge.scss (hunk 1 above)
Comment (Ap0c) [inline src/_shared/scss/_brandbadge.scss]: Do you mean `adverts__badge`? I can't find a use of `advert__badge` in the codebase...

File: src/_shared/js/capi-multiple.js [hunk 15]
```diff
@@ -0,0 +1,262 @@
...
+	}
+
+}
+
+// Constructs an individual card.
+function buildCard (cardInfo, cardNumber, isPaid) {
+
+	let adType = isPaid ? 'paidfor' : 'supported';
+	let cardTemplate = document.getElementById(`${adType}-card`);
+	let cardFragment = document.importNode(cardTemplate.content, true);
```
Comment (Ap0c) [inline src/_shared/js/capi-multiple.js]: Added a workaround in 540d51e.

**Comment by Ap0c** [comment]
I've switched the place where we handle IE9 (well, any browser that doesn't support `srcset`) over to the JS: d1b4d58. No need for the ugliness in `insertBetweenComments` any more 💥. @annebyrne @regiskuckaertz

File: src/_shared/js/capi-multiple.js [hunk 16]
```diff
@@ -111,34 +111,41 @@ function buildSources (sourceData) {
...
 
-// Insert between IE comments, a pain.
-function insertBetweenComments (card, sources) {
+// Creates a picture element with responsive sources, with fallback for IE.
+function createPicture (cardInfo, image) {
 
-	let picture = card.getElementsByTagName('picture')[0];
-	let pictureElems = picture.childNodes;
+	// Supports responsive images.
+	if (image.srcset === '') {
```
Comment (regiskuckaertz) [inline src/_shared/js/capi-multiple.js]: better: `'srcset' in image`
